addons/account/views/account_move_views.xml

```diff
@@ -63,7 +63,7 @@
                                         <field name="amount_currency"/>
                                     </group>
                                     <group string="Product" attrs="{'invisible': [('product_id', '=', False)]}">
-                                        <field name="product_id"/>
+                                        <field name="product_id" attrs="{'readonly':[('parent_state','=','posted')]}"/>
```

Comment:
I feel like this should be `readonly="1"` instead.
Same thing for all the other `attrs="{'readonly':[('parent_state','=','posted')]}"` on this view.

Editing these fields will almost always raise an error coming from the Journal Entry anyways. 
Changing the product will most likely change the debit/credit/taxes, possibly the account too. So it doesn't really make sense to change it.

If someone wants to change it, they are only one click away from the related `account.move`, where everything is under control in the form view.